src/diag.rs

```diff
@@ -194,6 +203,52 @@ where
     }
 }
 
+pub type StrWithHintResult<T> = Result<T, StrWithHint>;
+
+pub struct StrWithHint {
+    message: EcoString,
+    hints: Vec<EcoString>,
+}
+
+impl<T> At<T> for Result<T, StrWithHint> {
+    fn at(self, span: Span) -> SourceResult<T> {
+        self.map_err(|mut diags| {
+            Box::new(vec![
+                SourceError::new(span, diags.message).with_hints(&mut diags.hints)
+            ])
+        })
+    }
+}
+
+/// Allows adding a user-facing hint in addition to the error.
```

Comment:
Remark to the reviewer: Do you think this comment adds any value?

Reply:
I think it's fine. For me, even obvious comments help with understanding the code because I see that nothing non-obvious is going on.